src/components/FormikMovieForms/FormikAddMovie.js

```diff
@@ -0,0 +1,128 @@
+import React, { useState } from "react";
+import { Formik, Form, Field } from "formik";
+import * as Yup from "yup";
+import { useDispatch } from "react-redux";
+import {
+  addMovieActionCreator,
+  resetMovieActionCreator,
+} from "../../store/actionCreators";
+import { Link } from "react-router-dom";
+
+const MovieSchema = Yup.object().shape({
+  title: Yup.string()
+    .max(50, "The entry seems to be too long")
+    .required("This field is required"),
+
+  poster_path: Yup.string().required("This field is required"),
+
+  overview: Yup.string()
+    .min(20, "The overview needs to have at least 20 characters")
+    .required("This field is required"),
+
+  release_date: Yup.string().required("This field is required"),
+  genres: Yup.string().required("This field is required"),
+  runtime: Yup.number().required("This field is required"),
+});
+
+const FormikAddMovie = () => {
+  const [submitted, setSubmitted] = useState(false);
+  const dispatch = useDispatch();
+
+  const handleSubmit = (values) => {
+    setSubmitted(true);
+    values.genres = [values.genres];
+    values.id = Math.random().toString();
+    console.log(values);
+
+    dispatch(addMovieActionCreator(values));
+  };
+
+  const handleReset = (values) => {
+    values.genres = [values.genres];
+    values.title = "";
+  };
+
+  const fieldArray = [
+    { title: "Title", path: "title" },
+    { title: "Release Date", path: "release_date" },
+    { title: "Poster Path", path: "poster_path" },
+    { title: "Genre", path: "genres" },
+    { title: "Overview", path: "overview" },
+    { title: "Runtime", path: "runtime" },
+  ];
+
+  const FormikField = ({ title, path, myErrors, myTouched }) => {
```

Comment:
let's move it in separate file to reuse in EditMovie component

Reply:
Done